float32[] map
---
bool success